src/Bicep.MSBuild.E2eTests/src/example.ts

```diff
@@ -0,0 +1,72 @@
+
+import * as spawn from 'cross-spawn';
+import * as path from "path";
+import * as fs from "fs";
+import { SpawnSyncReturns } from 'node:child_process';
+
+export class Example {
+  readonly projectDir: string;
+  readonly projectFile: string;
+
+  constructor(projectName: string) {
+    const projectDir = path.normalize(path.join(__dirname, `../examples/${projectName}/`))
+      this.projectDir = projectDir,
+      this.projectFile = path.join(projectDir, `${projectName}.proj`)
+  }
+
+  public clean() {
+    const result = spawn.sync("git", ["clean", "-dfx", "--", this.projectDir], {
+      cwd: this.projectDir,
+      stdio: "pipe",
+      encoding: "utf-8",
+    });
+
+    if(result.status != 0) {
+      this.handleFailure("git", result);
+    }
+  }
+
+  public build(expectSuccess: boolean = true) {
+    const runtimeSuffix = process.env.RuntimeSuffix;
+    if(!runtimeSuffix) {
+      throw new Error("Please set the RuntimeSuffix environment variable to a .net runtime ID to run these tests. Possible values: win-x64, linux-x64, osx-x64");
+    }
+
+    const result = spawn.sync("dotnet", ["build", `/p:RuntimeSuffix=${runtimeSuffix}`, "/bl", this.projectFile], {
+      cwd: this.projectDir,
+      stdio: "pipe",
+      encoding: "utf-8",
+    });
+
+    if(expectSuccess && result.status != 0) {
```

Comment:
Convention is to use `===` and `!==` by default unless you've got very good reasons not to

Reply:
> Might be a bit late for this - but is packaging architecture-agnostic managed DLLs an option?
> 
> e.g. invoking it via:
> `dotnet ./src/Bicep.Cli/bin/Debug/net5.0/bicep.dll --help`
> 
> We'd need a check for dotnet >= 5 somehow (or could perhaps pull it down via nuget); which may or may not be an option for internal MSFT builds.

That was the initial approach I wanted to go with, but users with locked down pipelines do not always control which runtimes are installed on the build agents. I think in the future we can add a platform-agnostic version of the CLI nuget and update the MSBuild nuget to call `dotnet` on it, but the current CLI nugets will need to remain.

Reply:
The above reply went to the wrong comment, but fixed the wrong operator also.